### PR TITLE
libs/os: remove unused aliases, add test cases

### PR DESCRIPTION
**Pull request by alessio**
Remove unused ReadFile (unused) and
WriteFile (almost unused, alias of ioutil.WriteFile).

Add testcases for Must{Read,Write}File.

### CODE REVIEW

**Review by marbar3778 — COMMENTED** [review]
I love cleanup. Should we try to go one step further and see if we can remove this lib entirely?

**Comment by alessio** [comment]
Yep - happy to try get rid of it

File: libs/autofile/autofile_test.go [hunk 1]
```diff
@@ -147,3 +145,11 @@ func TestAutoFileSize(t *testing.T) {
 	// Cleanup
 	_ = os.Remove(f.Name())
 }
+
+func mustReadFile(t *testing.T, filePath string) []byte {
+	fileBytes, err := ioutil.ReadFile(filePath)
+
```
Comment (alexanderbez) [inline libs/autofile/autofile_test.go:151]:
```suggestion
```

File: config/toml.go [hunk 2]
```diff
@@ -492,16 +493,22 @@ func ResetTestRootWithChainID(testName string, chainID string) *Config {
...
+	mustWriteFile(privKeyFilePath, []byte(testPrivValidatorKey), 0644)
+	mustWriteFile(privStateFilePath, []byte(testPrivValidatorState), 0644)
 
 	config := TestConfig().SetRoot(rootDir)
 	return config
 }
 
+func mustWriteFile(filePath string, contents []byte, mode os.FileMode) {
+	if err := ioutil.WriteFile(filePath, contents, mode); err != nil {
+		tmos.Exit(fmt.Sprintf("MustWriteFile failed: %v", err))
```
Comment (alexanderbez) [inline config/toml.go:508]:
```suggestion
		tmos.Exit(fmt.Sprintf("failed to write file: %v", err))
```

**Review by melekes — APPROVED** [review]
👍

**Review by erikgrinaker — APPROVED** [review]
Great to see these go. Note that `libs/os` is currently designated a "supported" API in `README.md`, so these are breaking changes that need to be listed in the changelog.

**Comment by alessio** [comment]
@erikgrinaker Changelog amended. Thanks!

**Pull request merged**